addons/web/static/src/views/fields/state_selection/state_selection_field.js

```diff
@@ -1,39 +1,35 @@
 /** @odoo-module **/
 
-import { useCommand } from "@web/core/commands/command_hook";
+import { useService } from "@web/core/utils/hooks";
 import { Dropdown } from "@web/core/dropdown/dropdown";
 import { DropdownItem } from "@web/core/dropdown/dropdown_item";
 import { registry } from "@web/core/registry";
+import { sprintf } from "@web/core/utils/strings";
 import { _lt } from "@web/core/l10n/translation";
 import { standardFieldProps } from "../standard_field_props";
 import { formatSelection } from "../formatters";
 
-const { Component } = owl;
+const { Component, useEffect } = owl;
 
 export class StateSelectionField extends Component {
     setup() {
-        if (this.props.record.activeFields[this.props.name].viewType === "form") {
-            const commandName = this.env._t("Set kanban state...");
-            useCommand(
-                commandName,
-                () => {
-                    return {
-                        placeholder: commandName,
-                        providers: [
-                            {
-                                provide: () =>
-                                    this.options.map((value) => ({
-                                        name: value[1],
-                                        action: () => {
-                                            this.props.update(value[0]);
-                                        },
-                                    })),
-                            },
-                        ],
-                    };
-                },
-                { category: "smart_action", hotkey: "alt+shift+r" }
-            );
+        if (this.props.record.activeFields[this.props.name].viewType !== "form") {
+            const commandService = useService("command");
+            for (const [index, [value, label]] of this.options.entries()) {
+                const hotkey = String.fromCharCode(106 + index); // 'j', 'k', 'l'
+                useEffect(
+                    (dependencies) => {
+                        if (!dependencies || value === this.currentValue) return;
```

Comment:
"dependencies" is not a good name, if it is the currentValue, it should be named "currentValue", you should also not read it from "this", use the value that you get as the argument. This is especially important in the cleanup function (though in this case it doesn't matter) because it will be executed at some arbitrary later time, at which point the value on "this" may have changed but the argument value is captured in the closure and will not change.

Also files should be linted, see `/web/tooling/enable.sh`